### PR TITLE
(PUP-7911) Make 'plan' only be supported with --tasks feature

### PR DESCRIPTION
**Pull request by hlindberg**
This feature-switches the 'plan' keyword in the lexer and grammar so
that it is a warning to use it as a bare-word when feature is off, and
it is a token PLAN when feature is on.

### CODE REVIEW

File: lib/puppet/pops/parser/lexer2.rb [hunk 1]
```diff
@@ -138,15 +138,36 @@ class Lexer2
...
   }
 
   KEYWORDS.each {|k,v| v[1].freeze; v.freeze }
   KEYWORDS.freeze
 
+  # We maintain two different tables of tokens for the constructs
+  # introduced by application management. Which ones we use is decided in
+  # +initvars+; by selecting one or the other variant, we select whether we
+  # hit the appmgmt-specific code paths
+  TASK_MANAGEMENT_TOKENS = {
```
Comment (thallgren) [inline lib/puppet/pops/parser/lexer2.rb]: The comment mentions "application management" which seems a bit off.

File: lib/puppet/pops/parser/egrammar.ra [hunk 2]
```diff
@@ -24,6 +24,7 @@ token TYPE
 token PRIVATE ATTR
 token APPLICATION PRODUCES CONSUMES SITE
 token PLAN
+tokan PLAN_R
```
Comment (thallgren) [inline lib/puppet/pops/parser/egrammar.ra]: misspelled "tokan"

File: spec/unit/pops/parser/lexer2_spec.rb [hunk 3]
```diff
@@ -98,13 +98,27 @@ def epp_tokens_scanned_from(s)
...
+  context 'when --tasks' do
+    before(:each) { Puppet[:tasks] = true }
+    after(:each) { Puppet[:tasks] = false }
+
+    it "should lex a keyword from 'plan'" do
+      expect(tokens_scanned_from('plan')).to match_tokens2(:PLAN)
+    end
+  end
+
   # TODO: Complete with all edge cases
```
Comment (thallgren) [inline spec/unit/pops/parser/lexer2_spec.rb]: A test is probably missing given that `token` was  misspelled above. Not sure how that could work?
